src/github.com/mozilla-services/pushgo/simplepush/socket.go

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package simplepush
+
+import (
+	"bytes"
+	"encoding/json"
+	"time"
+
+	"golang.org/x/net/websocket"
+)
+
+type Socket interface {
```

Comment:
Doc strings? This is all for what purpose? I can guess its for mocking capability, but it'd be nice to doc these things so we remember in the future.